File: src/lib/software/java_sdk_21.py
from src.lib.software import BaseSoftware, SoftwareCategory

class JavaSEDevelopmentKit21(BaseSoftware):
    def __init__(self):
        super().__init__()

        self.key = 'java-sdk-21'
        self.name = 'Java SE Development Kit 21.x'
        self.category = SoftwareCategory.Development
        self.download_name = 'jdk-21_windows-x64_bin.msi'
        self.is_archive = False
        self.should_cache_url = False
        self.requires_admin = False
        self.icon = 'java.png'
        self.homepage = 'https://oracle.com/java/technologies/downloads'

    def resolve_download_url(self):
        self.url_resolved.emit('https://download.oracle.com/java/21/latest/jdk-21_windows-x64_bin.msi')

File: packages/app/src/services/software/definitions/javaSdk21.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'java-sdk-21';
	public name = 'Java SE Development Kit 21.x';
	public category = [
		SoftwareCategory.Development,
		SoftwareCategory.Runtime,
	];
	public downloadName = 'jdk-21_windows-x64_bin.msi';
	public isArchive = false;
	public shouldCacheUrl = false;
	public requiresAdmin = false;
	public deprecated = false;
	public icon = 'java.png';
	public homepage = 'https://oracle.com/java/technologies/downloads';

	public async resolveDownloadUrl() {
		return ok('https://download.oracle.com/java/21/latest/jdk-21_windows-x64_bin.msi');
	}
}
